sdk/python/kfp/_client.py

```diff
@@ -295,58 +296,150 @@ def list_pipelines(self, page_token='', page_size=10, sort_by=''):
     """
     return self._pipelines_api.list_pipelines(page_token=page_token, page_size=page_size, sort_by=sort_by)
 
+
   # TODO: provide default namespace, similar to kubectl default namespaces.
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, namespace=None):
-    """Run a specified pipeline.
+  def _create_job_config(self, experiment_id, params, pipeline_package_path, pipeline_id, namespace):
+    """Create a specified job.
 
     Args:
-      experiment_id: The string id of an experiment.
-      job_name: name of the job.
-      pipeline_package_path: local path of the pipeline package(the filename should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).
-      params: a dictionary with key (string) as param name and value (string) as as param value.
-      pipeline_id: the string ID of a pipeline.
-      namespace: kubernetes namespace where the pipeline runs are created.
+    experiment_id: The string id of an experiment.
+    pipeline_package_path: Local path of the pipeline package(the filename
+        should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).
+    params: A dictionary with key (string) as param name and value (string)
+        as as param value.
+    pipeline_id: The string ID of a pipeline.
+    namespace: Kubernetes namespace where the pipeline runs are created.
         For single user deployment, leave it as None;
         For multi user, input a namespace where the user is authorized
 
     Returns:
-      A run object. Most important field is id.
+    A JobConfig object with attributes spec and resource_reference.
     """
+    class JobConfig:
+      def __init__(self, spec, resource_references):
+        self.spec = spec
+        self.resource_references = resource_references
 
     pipeline_json_string = None
     if pipeline_package_path:
       pipeline_obj = self._extract_pipeline_yaml(pipeline_package_path)
       pipeline_json_string = json.dumps(pipeline_obj)
-    api_params = [kfp_server_api.ApiParameter(
+    api_params = [
+      kfp_server_api.ApiParameter(
         name=sanitize_k8s_name(name=k, allow_capital_underscore=True),
-        value=str(v)) for k,v in params.items()]
+        value=str(v),
```

Comment:
also this shouldn't be applied. you might have to turn off auto formatting and just do it yourself